src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspector.php

```diff
@@ -240,8 +253,17 @@ public function getLocalizedUrlsForPage(BasePageDocument $page)
         $rlpProperty = $structure->getPropertyByTagName('sulu.rlp');
 
         foreach ($webspace->getAllLocalizations() as $localization) {
-            $locale = $localization->getLocalization();
-            $stageName = $this->encoder->localizedSystemName(WorkflowStageSubscriber::WORKFLOW_STAGE_FIELD, $locale);
+            $resolvedLocale = $localization->getLocalization();
+            $locale = $resolvedLocale;
+
+            $shadowEnabledName = $this->encoder->localizedSystemName(ShadowLocaleSubscriber::SHADOW_ENABLED_FIELD, $resolvedLocale);
+
+            if (true === $node->getPropertyValueWithDefault($shadowEnabledName, false)) {
+                $shadowLocaleName = $this->encoder->localizedSystemName(ShadowLocaleSubscriber::SHADOW_LOCALE_FIELD, $resolvedLocale);
+                $resolvedLocale = $node->getPropertyValue($shadowLocaleName);
+            }
+
+            $stageName = $this->encoder->localizedSystemName(WorkflowStageSubscriber::WORKFLOW_STAGE_FIELD, $resolvedLocale);
```

Comment:
We need a better way to know if a document is translated in a given locale or not.